cherab/generomak/machine/plasma.py

```diff
@@ -0,0 +1,213 @@
+
+    :return: Dictionary holding instances of Maxwellian distributions for plasma species.
+    """
+
+    mesh_interp = get_edge_interpolators()
+
+    zero_vector = Vector3D(0, 0, 0)
+
+    dists = RecursiveDict()
+
+    n3d = AxisymmetricMapper(mesh_interp["electron"]["density"])
+    t3d = AxisymmetricMapper(mesh_interp["electron"]["temperature"])
+
+    dists["electron"] = Maxwellian(n3d, t3d, zero_vector, electron_mass)
+
+    for elem_name, elem_data in mesh_interp["composition"].items():
+        for stage, stage_data in elem_data.items():
+            
+            n3d = AxisymmetricMapper(stage_data["density"])
+            t3d = AxisymmetricMapper(stage_data["temperature"])
+            mass = stage_data["element"].atomic_weight * atomic_mass
+            dists["composition"][elem_name][stage]["distribution"] = Maxwellian(n3d, t3d, zero_vector, mass)
+            dists["composition"][elem_name][stage]["element"] = stage_data["element"]
+
+    return dists.freeze()
+
+def get_edge_plasma(atomic_data=None):
+    """
+    Provides Generomak Edge plasma.
+
+    :param atomic_data: Instance of AtomicData, default isOpenADAS()
+    :return: populated Plasma object
+    """
+    
+    #create or check atomic_data
+    if atomic_data is not None:
+     if not isinstance(atomic_data, AtomicData):
+         raise ValueError("atomic_data has to be of type AtomicData")   
+    else:
+        atomic_data = OpenADAS()
+
+    # base plasma geometry on mesh vertices
+    profiles_dir = os.path.join(os.path.dirname(__file__), "data/plasma/edge")
+    path = os.path.join(profiles_dir, "mesh.json")
+    with open(path, "r") as fhl:
+        mesh = json.load(fhl)
+
+    vertex_coords = np.asarray(mesh["vertex_coords"])
+    r_range = (vertex_coords[:, 0].min(), vertex_coords[:, 0].max())
+    z_range = (vertex_coords[:, 1].min(), vertex_coords[:, 1].max())
+    plasma_height = z_range[1] - z_range[0]
+
+    padding = 1e-3 #enlarge plasma geometry for safety
+
+    outer_column = Cylinder(radius=r_range[1] + padding, height=plasma_height + 2 * padding)
+    inner_column = Cylinder(radius=r_range[0] - padding, height=plasma_height + 4 * padding)
+    inner_column.transform = translate(0, 0, -2 * padding)
+
+    plasma_geometry = Subtract(outer_column, inner_column)
+    geometry_transform = translate(0, 0, -outer_column.height / 2)
+    
+    # get distributions
+    dists = get_edge_distributions()
+
+    # create plasma composition list
+    plasma_composition = []
+    for elem_name, elem_data in dists["composition"].items():
+        for stage, stage_data in elem_data.items():
+            species = Species(stage_data["element"], stage, stage_data["distribution"])
+            plasma_composition.append(species)
+
+    # Populate plasma
+    plasma = Plasma()
+    plasma.geometry = plasma_geometry
+    plasma.atomic_data = atomic_data
+    plasma.electron_distribution = dists["electron"]
+    plasma.composition = plasma_composition
```

Comment:
Since we already have equilibrium, is it possible to add `b_field` from there?

Reply:
Good point. Added b_field

Reply:
Thanks, the magnetic field will be needed for high-resolution spectroscopy demos.